packages/gatsby/src/utils/webpack-utils.ts

```diff
@@ -512,11 +512,14 @@ export const createWebpackUtils = (
         `warning`,
         `webpack`,
       ]
-      const doNotPolyfillRegex = new RegExp(
-        `[\\\\/]node_modules[\\\\/](${VENDORS_TO_NOT_POLYFILL.join(
-          `|`
-        )})[\\\\/]`
-      )
+
+      const doNotPolyfillRegex = hasIESupport(process.cwd())
```

Comment:
As per making moves, we'll want to move this `hasIESupport` check up one level and don't even use the loader if not applicable